### PR TITLE
Consenus tests for Conway era. Extended Test.Cardano.Ledger.Generic.Proof to Conway Era

### LINKED ISSUE #2947: Barebones conway test package
- [ ] Create a new cabal package for the conway ledger era test suite
- [ ] Create the conway example for consensus. The [babbage example](https://github.com/input-output-hk/cardano-ledger/blob/dd4b6e31a05e3c02c6f00f0d53cf20e3b8b74468/eras/babbage/test-suite/src/Test/Cardano/Ledger/Babbage/Examples/Consensus.hs) can be re-used for this.
- [ ] Extend the generic testing framework in `cardano-ledger-tests` to work in the conway era.

### LINKED ISSUE #2968: conway CDDL spec
The conway CDDL spec may very well be identical to the babbage spec. We do not need to copy the spec (in my opinion, but I am open to opinions from others), but we should make it clear somehow that the conway wire spec is the babbage wire spec (for as long as this holds true).

### LINKED ISSUE #2969: Run existing era-generic tests in conway
The work to extend the era-generic framework to conway was done in  #2967, but the tests are not actually being run in conway yet (neither the unit tests / examples, or the property tests). Add the tests to our test suite.

### PR DESCRIPTION
**Pull request by TimSheard**
Filed out some missing pieces for the Conway era.
1) The Consenus tests
2) The Generic tests in Test.Cardano.Ledger.Generic.
3) Added to what it means to be (GoodCrypto era) in Test.Cardano.Ledger.Generic.Proof
4) Added Conway unit tests (TwoPhaseValidation BabbageFeatures) 
5) Added Conway property tests

resolves #2968
resolves #2969
resolves #2947

### CODE REVIEW

File: eras/conway/impl/cardano-ledger-conway.cabal [hunk 1]
```diff
@@ -48,18 +48,17 @@ library
   build-depends:
                 bytestring,
                 cardano-binary,
-                cardano-crypto-class,
+                -- cardano-crypto-class,
```
Comment (JaredCorduan) [inline eras/conway/impl/cardano-ledger-conway.cabal]: if these are not needed, we should delete them

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/GenState.hs [hunk 2]
```diff
@@ -887,7 +892,8 @@ genPlutusScript proof tag = do
       corescript = case proof of
         Alonzo _ -> script
         Babbage _ -> script
-        _ -> error ("Only Alonzo and Babbage have PlutusScripts. " ++ show proof ++ " does not.")
+        Conway _ -> script
+        _ -> error ("Only Alonzo, Babbage, and Conway have PlutusScripts. " ++ show proof ++ " does not.")
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/GenState.hs]: maybe we could keep this from rotting by saying "PlutusScripts are available starting in the Alonzo era".

**Review by JaredCorduan — APPROVED** [review]
Looks great! I just made a few minor comments.

@TimSheard and I talked about how we will clean this up later (reducing the code copy). But for now we want to unblock the consensus work to support `conway`.

File: eras/conway/impl/cardano-ledger-conway.cabal (hunk 1 above)
Comment (TimSheard) [reply eras/conway/impl/cardano-ledger-conway.cabal]: done

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/GenState.hs (hunk 2 above)
Comment (TimSheard) [reply libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/GenState.hs]: done

File: eras/conway/test-suite/cardano-ledger-conway-test.cabal [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+author:              IOHK Formal Methods Team
+maintainer:          formal.methods@iohk.io
+copyright:           2020 Input Output (Hong Kong) Ltd.
+category:            Network
+build-type:          Simple
+
+data-files:
+  cddl-files/conway.cddl
+  cddl-files/real/crypto.cddl
+  cddl-files/mock/extras.cddl
```
Comment (JaredCorduan) [inline eras/conway/test-suite/cardano-ledger-conway-test.cabal]: These files do not exist yet. We are expecting the conway wire spec to be identical to the babbage wire spec, so we probably do not even want them to exist (except perhaps for consistency). :thinking:

Comment (TimSheard) [reply eras/conway/test-suite/cardano-ledger-conway-test.cabal]: So, should these lines be deleted?

Comment (JaredCorduan) [reply eras/conway/test-suite/cardano-ledger-conway-test.cabal]: yes!

Comment (TimSheard) [reply eras/conway/test-suite/cardano-ledger-conway-test.cabal]: So they are now deleted

**Pull request merged**